python/pyiceberg/table/metadata.py

```diff
@@ -28,11 +30,11 @@
 
 from pydantic import Field
 from pydantic import ValidationError as PydanticValidationError
-from pydantic import root_validator
+from pydantic import model_validator
 from typing_extensions import Annotated
 
 from pyiceberg.exceptions import ValidationError
-from pyiceberg.partitioning import PartitionSpec, assign_fresh_partition_spec_ids
+from pyiceberg.partitioning import PARTITION_FIELD_ID_START, PartitionSpec, assign_fresh_partition_spec_ids
```

Comment:
Ah, so this was needed outside of `partitioning` and that's why it is no longer prefixed with an underscore?

Reply:
Exactly, the `_` means that it shouldn't be used outside of the class.